### ISSUE TITLE
Expected printf output from shell, but got: "bash: cannot set terminal process group (-1) [...]"

### ISSUE THREAD

**Issue by mmxgn** [millejoh/emacs-ipython-notebook#718]
## Problem description
Hi,

Ein is one of the coolest packages out there. Thanks to everyone maintaining it!

I freshly installed it wanting to give another try into turning emacs into my main driver. When I try to run a python notebook (or even when I am reading it with `C-x f`) I get the following error:

```cl
File mode specification error: (error Expected printf output from shell, but got: "bash: cannot set terminal process group (-1): Inappropriate ioctl for device
bash: no job control in this shell
bash debugger, bashdb, release 5.0-1.1.2

Copyright 2002-2004, 2006-2012, 2014, 2016-2019 Rocky Bernstein
This is free software, covered by the GNU General Public License, and you are
welcome to change it and/or distribute copies of it under certain conditions.

bash: _Dbg_filenames[\"$fullname\"]: bad array subscript
")
user-error: Minibuffer window is not active
Mark set

```

## Steps to reproduce the problem

1. Open a `.ipynb` file.
2. Type `ein:run`

## System info:

```cl
("EIN system info"
 :emacs-version
 "GNU Emacs 26.3 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.20)
 of 2020-05-19"
 :emacs-bzr-version nil
 :window-system x
 :emacs-variant nil
 :os
 (:uname
  "Linux emerdesktop 5.6.14-arch1-1 #1 SMP PREEMPT Wed, 20 May 2020 20:43:19 +0000 x86_64 GNU/Linux
"
  :lsb-release
  "LSB Version:	1.4
Distributor ID:	Arch
Description:	Arch Linux
Release:	rolling
Codename:	n/a
")
 :notebook
 "Name: notebook
Version: 6.0.3
Summary: A web-based notebook environment for interactive computing
Home-page: http://jupyter.org
Author: Jupyter Development Team
Author-email: jupyter@googlegroups.com
License: BSD
Location: /home/mmxgn/anaconda3/envs/tf_gpu/lib/python3.7/site-packages
Requires: traitlets, tornado, jinja2, pyzmq, ipykernel, terminado, jupyter-client, jupyter-core, nbformat, prometheus-client, nbconvert, ipython-genutils, Send2Trash
Required-by: widgetsnbextension, jupyterlab, jupyterlab-server, jupyter, hide-code
"
 :ipython "7.13.0
"
 :image-types
 (svg imagemagick png gif tiff jpeg xpm xbm pbm postscript)
 :image-types-available
 (svg imagemagick png gif tiff jpeg xpm xbm pbm postscript)
 :request
 (:backend curl)
 :ein
 (:version "20200514.2040"
	   :source-dir "/home/mmxgn/.emacs.d/elpa/ein-20200514.2040/")
 :lib
 ((:name "websocket"
	 :path "~/.emacs.d/elpa/websocket-20200419.2124/websocket.elc"
	 :featurep t
	 :version-var websocket-version
	 :version "1.12")
  (:name "anaphora"
	 :path "~/.emacs.d/elpa/anaphora-20180618.2200/anaphora.elc"
	 :featurep t
	 :version-var nil
	 :version nil)
  (:name "request"
	 :path "~/.emacs.d/elpa/request-20200517.1305/request.elc"
	 :featurep t
	 :version-var request-version
	 :version "0.3.0")
  (:name "deferred"
	 :path "~/.emacs.d/elpa/deferred-20170901.1330/deferred.elc"
	 :featurep t
	 :version-var deferred:version
	 :version "0.5.0")
  (:name "polymode"
	 :path "~/.emacs.d/elpa/polymode-20200411.915/polymode.elc"
	 :featurep t
	 :version-var nil
	 :version nil)
  (:name "exec-path-from-shell"
	 :path "~/.emacs.d/elpa/exec-path-from-shell-20191229.112/exec-path-from-shell.elc"
	 :featurep t
	 :version-var nil
	 :version nil)
  (:name "dash"
	 :path "~/.emacs.d/elpa/dash-20200426.2244/dash.elc"
	 :featurep t
	 :version-var nil
	 :version nil)))
```

**Comment by dickmao** [comment]
If a simple `C-x C-f` produces this bad interaction with Rocky Bernstein's bashdb, then it's independent of EIN.  As described in our README, `emacs -Q -f package-initialize --eval "(setq debug-on-error t)"` is the only way to know for sure.

**Comment by mmxgn** [comment]
Hi again,
I had also tried with `emacs -Q -f package-initialize --eval "(setq debug-on-error t)"` and the problem is the same, here is the more verbose output:

```cl
Debugger entered--Lisp error: (error "Expected printf output from shell, but got: \"bash: cannot set terminal process group (-1): Inappropriate ioctl for device\nbash: no job control in this shell\nbash debugger, bashdb, release 5.0-1.1.2\n\nCopyright 2002-2004, 2006-2012, 2014, 2016-2019 Rocky Bernstein\nThis is free software, covered by the GNU General Public License, and you are\nwelcome to change it and/or distribute copies of it under certain conditions.\n\nbash: _Dbg_filenames[\\\"$fullname\\\"]: bad array subscript\n\"")
  signal(error ("Expected printf output from shell, but got: \"bash: cannot set terminal process group (-1): Inappropriate ioctl for device\nbash: no job control in this shell\nbash debugger, bashdb, release 5.0-1.1.2\n\nCopyright 2002-2004, 2006-2012, 2014, 2016-2019 Rocky Bernstein\nThis is free software, covered by the GNU General Public License, and you are\nwelcome to change it and/or distribute copies of it under certain conditions.\n\nbash: _Dbg_filenames[\\\"$fullname\\\"]: bad array subscript\n\""))
  error("Expected printf output from shell, but got: %S" "bash: cannot set terminal process group (-1): Inappropriate ioctl for device\nbash: no job control in this shell\nbash debugger, bashdb, release 5.0-1.1.2\n\nCopyright 2002-2004, 2006-2012, 2014, 2016-2019 Rocky Bernstein\nThis is free software, covered by the GNU General Public License, and you are\nwelcome to change it and/or distribute copies of it under certain conditions.\n\nbash: _Dbg_filenames[\"$fullname\"]: bad array subscript\n")
  exec-path-from-shell-printf("%s\\000%s" ("${PATH-32b9c535d6ce48bbf03183935c987340}" "${MANPATH-32b9c535d6ce48bbf03183935c987340}"))
  exec-path-from-shell-getenvs(("PATH" "MANPATH"))
  exec-path-from-shell-copy-envs(("PATH" "MANPATH"))
  exec-path-from-shell-initialize()
  #f(compiled-function (symbol value) #<bytecode 0x10f6af1>)(ein:jupyter-server-command "jupyter")
  custom-initialize-reset(ein:jupyter-server-command "jupyter")
  custom-declare-variable(ein:jupyter-server-command "jupyter" "The default command to start a jupyter notebook server.\n\nChanging this to `jupyter-notebook' requires customizing `ein:jupyter-server-use-subcommand' to nil." :group ein :type string :set #f(compiled-function (symbol value) #<bytecode 0x10f6af1>))
  byte-code("\306\307!\210\306\310!\210\306\311!\210\306\312!\210\313\314\315\316\317\320\321\322&\007\210\313\323\324\325\317\320\321\326&\007\210\313\327\330\331\317\320\321\326&\007\210\313\332\333\334\317\320\321\326&\007\210\313\301\335\336\317\320\321\326\337\340&\011\210\313\341\301\342\317\320\321\326\343\344\337\345&\013\210\313\346\347\350\317\320\321\351&\007\210\313\352\353\354\317\320\321\355&\007\210\313\356\315\357\317\320\321\360&\007\210\313\361\362\363\317\320\321\364\365\3661\310\0\367\370\371\372\373\030\374\375\376\377\011\"!!)\"\032\315\211\033\034\315\035\n:\203\300\0\n@\211\023\211A\023\242\211\024\371\201A\0\371\303\013\"\"B\015B\025\nA\211\022\202\236\0\015\237,\"0\202\331\0\036@\201B\0\201C\0\201D\0\"\210)\201E\0\"&\007\207" [json-object-type ein:jupyter-server-command --cl-var-- spec k --cl-var-- require ein-core ein-notebooklist ein-dev exec-path-from-shell custom-declare-variable ein:jupyter-use-containers nil "Take EIN in a different direcsh." :group ein :type boolean ein:jupyter-docker-image "jupyter/datascience-notebook" "Docker pull whichever jupyter image you prefer.  This defaults to\nthe 'jupyter docker stacks' on hub.docker.com.\n\nOptionally append ':tag', e.g., ':latest' in the customary way." string ein:jupyter-docker-mount-point "/home/jovyan/ipynb" "Directory in docker image where to mount `ein:jupyter-default-notebook-directory'." ein:jupyter-docker-additional-switches "-e JUPYTER_ENABLE_LAB=no --rm" "Additional options to the 'docker run' call.\n\nNote some options like '-v' and '-network' are imposed by EIN." "jupyter" "The default command to start a jupyter notebook server.\n\nChanging this to `jupyter-notebook' requires customizing `ein:jupyter-server-use-subcommand' to nil." :set #f(compiled-function (symbol value) #<bytecode 0x10f6af1>) ein:jupyter-default-server-command "Obsolete alias for `ein:jupyter-server-command'" :set-after (ein:jupyter-server-command) #f(compiled-function (_symbol value) #<bytecode 0x10cb3bd>) ein:jupyter-server-use-subcommand "notebook" "Users of \"jupyter-notebook\" (as opposed to \"jupyter notebook\") need to Omit." (choice (string :tag "Subcommand" "notebook") (const :tag "Omit" nil)) ein:jupyter-server-args (quote ("--no-browser")) "Add any additional command line options you wish to include\nwith the call to the jupyter notebook." (repeat string) ein:jupyter-default-notebook-directory "Default location of ipynb files." directory ein:jupyter-default-kernel ...] 19)
  require(ein-jupyter)
  eval-buffer(#<buffer  *load*-686335> nil "/home/mmxgn/.emacs.d/elpa/ein-20200514.2040/ein-process.el" nil t)  ; Reading at buffer position 927
  load-with-code-conversion("/home/mmxgn/.emacs.d/elpa/ein-20200514.2040/ein-process.el" "/home/mmxgn/.emacs.d/elpa/ein-20200514.2040/ein-process.el" nil t)
  require(ein-process)
  eval-buffer(#<buffer  *load*> nil "/home/mmxgn/.emacs.d/elpa/ein-20200514.2040/ein-ipynb-mode.el" nil t)  ; Reading at buffer position 903
  load-with-code-conversion("/home/mmxgn/.emacs.d/elpa/ein-20200514.2040/ein-ipynb-mode.el" "/home/mmxgn/.emacs.d/elpa/ein-20200514.2040/ein-ipynb-mode.el" nil t)
  ein:ipynb-mode()
  set-auto-mode-0(ein:ipynb-mode nil)
  set-auto-mode()
  normal-mode(t)
  after-find-file(nil t)
  find-file-noselect-1(#<buffer Visualisation.ipynb> "~/Projects/Public/tutorials/pytorch/Visualisation.ipynb" nil nil "~/Projects/Public/tutorials/pytorch/Visualisation.ipynb" (114300277 2052))
  find-file-noselect("~/Projects/Public/tutorials/pytorch/Visualisation.ipynb" nil nil t)
  find-file("~/Projects/Public/tutorials/pytorch/Visualisation.ipynb" t)
  funcall-interactively(find-file "~/Projects/Public/tutorials/pytorch/Visualisation.ipynb" t)
  call-interactively(find-file nil nil)
  command-execute(find-file)
```
Could you please direct me to where should I search to find the culprit if that's not EIN-related?

**Comment by dickmao** [comment]
Oh my apologies then.  Commit 21adeb9 should avoid this error.  You might have to wait an hour for the melpa rebuild before reinstalling the package.

**Comment by mmxgn** [comment]
Hi,

This fixed it! Thanks greatly!
